### ISSUE TITLE
tf.gradients should return an error if the xs list is empty

### ISSUE THREAD

**Issue by Andrew-Draganov** [tensorflow/tensorflow#21103]
You can currently call tf.gradients(ys, xs) where xs is an empty list. The returned result is None. I feel like it should return an error, though, to keep in line with the definition of the gradient -- dx/dy is undefined if x is undefined. 

Example code:

    a = tf.Variable([100])
    b = tf.reduce_sum(a)
    grads = tf.gradients(b, [])
    with tf.Session() as sess:
        print(sess.run(grads))

`None`

This is somewhat related to issue #783 (Gradient computation erroneously returns None), at least in the sense that the tf.gradients implementation deviates from the mathematical definition. I also understand that there may be downstream complications that make this unreasonable to implement. I looked around for an issue regarding this but couldn't find one... sorry if I missed anything!

EDIT: filling out template
Platform - Ubuntu 16.04
Tensorflow installed with pip3 in venv
Tensorflow version 1.8.0
N/A for the rest (they don't really apply to the question)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 35 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 50 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 65 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 95 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 110 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
